### PR TITLE
{rad,son,baz,lid,read,recycl}arr: add mainProgram

### PR DESCRIPTION
**Pull request by SuperSamus**
## Description of changes

Doesn't hurt to add it.

One question though: should we keep Sonarr's old name `NzbDrone`? The [AUR package](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=sonarr) for instance doesn't use it.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/radarr/default.nix [hunk 1]
```diff
@@ -52,6 +52,7 @@ in stdenv.mkDerivation rec {
     changelog = "https://github.com/Radarr/Radarr/releases/tag/v${version}";
     license = licenses.gpl3Only;
     maintainers = with maintainers; [ edwtjo purcell ];
+    mainProgramm = "Radarr";
```
Comment (MikaelFangel) [inline pkgs/servers/radarr/default.nix:55]:
```suggestion
    mainProgram = "Radarr";
```

**Comment by ambroisie** [comment]
I think renaming it from `NzbDrone` is a breaking change, so that would need to wait until after the 23.11 window.

Though one could argue that outside of the NixOS module there are probably very few uses to actually break...

**Review by ambroisie — APPROVED** [review]

**Comment by ambroisie** [comment]
What about the other *arr BTW?

**Comment by SuperSamus** [comment]
Never used them, so I don't know their executable name. (Is it just their name with the first letter capitalized?)

**Comment by ambroisie** [comment]
They appear in the package, since they all need to be wrapped to run under `mono`.

But yes that's what I would assume.

**Review by ambroisie — APPROVED** [review]

**Pull request merged**